pkg/taxonomy/schema.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package taxonomy
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+
+	"emperror.dev/errors"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/mesh-for-data/openapi2crd/pkg/generator"
+	"github.com/xeipuuv/gojsonschema"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ValidateSchema validates that the input schema adheres to the requirements defined in
+// https://github.com/mesh-for-data/mesh-for-data/blob/master/config/taxonomy/HOWTO_SPEC.md
+func ValidateSchema(path string) error {
+	orig := path
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return err
+	}
+	path = filepath.Clean(path)
+
+	err = ValidateDraft4(path)
+	if err != nil {
+		return errors.Wrap(err, "Schema is not a valid DRAFT 4 JSON schema")
+	}
+
+	err = ValidateStructural(orig)
```

Comment:
Why orig? What happened when you used path?
Using a filepath without cleaning it first is considered a security risk, when we start using something like gosec it will capture those as errors (which reminds me to open an issue for it).

Reply:
+1